portfolio/src/main/webapp/script.js

```diff
@@ -114,19 +115,300 @@ function createCommentElement(server_comment) {
  * @param comment: a FormSubmissions instance stored in the server
  */
 function deleteComment(comment) {
-    const params = new URLSearchParams();
-    params.append('id', comment.id);
-    fetch('/delete-comment', {method: 'POST', body:params});
+  const params = new URLSearchParams();
+  params.append('id', comment.id);
+  fetch('/delete-comment', {method: 'POST', body:params});
 }
 
-/** Tells the server to fetch and delete all comments. */
+/** 
+ * Tells the server to fetch and delete all comments. 
+ */
 function deleteAllComments() {
-    fetch('/data').then(response => response.text()).then((comments) => {
-      const commentsElement = document.getElementById('comments-container');
-      const commentsObj = JSON.parse(comments);
-      commentsObj.forEach((comment) => {
-        deleteComment(comment);
-      })
-      commentsElement.innerText = '';
+  fetch('/data').then(response => response.text()).then((comments) => {
+    const commentsElement = document.getElementById('comments-container');
+    const commentsObj = JSON.parse(comments);
+    commentsObj.forEach((comment) => {
+      deleteComment(comment);
+    })
+    commentsElement.innerText = '';
+  });
+}
+
+// Initialize variables.
+var map;
+// Create default lat and lng variables for lat, lng points of Google LAX office.
+var defaultLat= 33.9955566;
+var defaultLng = -118.4768693;
+// Create editable marker that displays when user clicks in the map.
+var userMarker;
+
+/**
+ * Creates a map and adds it to the page.
+ */
+function initMap() {
+
+  // Call function to create purple styled map.
+  createStyledMap();
+  
+  // Create map.
+  map = new google.maps.Map(document.getElementById('map'), {
+    center: {lat: defaultLat, lng: defaultLng},
+    zoom: 10,
+    mapTypeControlOptions: {
+     mapTypeIds: ['roadmap', 'satellite', 'terrain', 'styled_map']
+    }
+  });
+
+  // Allow users to add markers of their own by clicking on map to show a marker a with text box the user can edit. 
+  map.addListener('click', (event) => {
+    createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+  });
+
+  // Call function that fetches data of locations I have visited to add to map.
+  fetchMyMarkers();
+  // Call function that fetches data of locations users have visited to add to map.
+  fetchUserMarkers();
+
+  // Associate the purple styled map with the MapTypeId and set it to display.
+  map.mapTypes.set('styled_map', purpleStyledMap);
+  map.setMapTypeId('styled_map');
+}
+
+/**
+ * Fetches the data representing locations I have visited from the server.
+ */
+function fetchMyMarkers() {
```

Comment:
fetchMyMarkers/fetchUserMarkers and createMyMarkers/createUserMarkers are very similar. Consider refactor the code to combine them, and maybe pass in a bool to differentiate the logic.